### DISCUSSION TITLE
Предложения по движковой боевке

### DISCUSSION THREAD

**Discussion by 1Pixel1** [OGSR/OGSR-Engine#283] [Ideas]
1. В оригинальной игре, если НПС теряют из виду цель, то сбиваются в кучу, стоят какое-то время, орут, а затем расходятся. Хотелось бы, чтобы они более адекватно вели себя в этой ситуации.
https://i.postimg.cc/L8r473RF/1.jpg
На скрине ГГ находится за деревом, отмеченным красным. Если я правильно понимаю, то это недоработки движковой боевки. (Скрин с OGSR engine, но в оригинале тоже самое, проверено).

2. При стрельбе по НПС они часто стоят на месте, даже если получили урон из оружия, не говоря уже про попадание рядом с ними.

3. Возможно алгоритм для "Обходи эту шелупонь!" тоже стоит доработать.

Про стелс:

4. В AI additions 2 есть правка с которой у НПС, погибшего от выстрела/удара в голову, не воспроизводится предсмертный крик. Там это сделано скриптами, возможно более правильно сделать это в движке.

5. Хотелось бы иметь возможность изменить в конфиге коэффициент ослабления слышимости звуков шагов ГГ при медленной ходьбе и в присяде. Чтобы можно было настроить возможность тихо подойти к НПС и в тоже время не менять звуки, чтобы не трогать слышимость шагов ГГ при беге.

**Comment by xrSimpAI** [comment]
Если говорить о багах движковой боевки / аи нпс, их значительно больше)
Да и сомневаюсь что здесь кто-то будет в этом возиться, это оч муторная работа, я так вообще в движке в этом году особо ничего не делаю, как-то это всё надоело. Проще поставить скриптовые аи аддишенсы всякие (естественно последние версии, а не за 2009 год, как в большинстве модов), в чем-то это ситуацию улучшит ( конечно не значительно, но всё же ).

**Comment by joye-ramone** [comment]
5 кстати можно сделать довольно просто
Но без скриптовых костылей стелс в сталкере все равно будет странный и убогий.

**Reply by 1Pixel1** [reply]
Стелс убогий, но если хотя бы пункты 4, 5 добавить, то будет уже лучше (и еще пошаманить с m_stalker). По крайней мере весь лагерь не будет подскакивать, если одному неписю в дыню зарядить. В оригинале у звуков предсмертных криков радиус слышимости очень большой. Понятно, что можно этот радуис уменьшить, но для других ситуаций, отличных от хедшота, это решение уже не очень. Хорошо бы, если была возможность пофиксить это как я писал выше или ниже.

**Comment by 1Pixel1** [comment]
пункт 4
Возможно ли добавить передачу bone_index в function motivator_binder:death_callback(victim, who) по аналогии с function motivator_binder:hit_callback(obj, amount, local_direction, who, bone_index)
Тогда по идее можно будет по индексу кости определять прилетело ли неписю в башку и если так, то отключать его предсмертный звук npc:set_sound_mask(-1)
